aiida/backends/tests/cmdline/commands/test_computer.py

```diff
@@ -1,13 +1,357 @@
+
+    if non_interactive:
+        valid_noninteractive_options['non-interactive'] = None
+    valid_noninteractive_options['label'] = 'noninteractive_computer'
+    valid_noninteractive_options['hostname'] = "localhost"
+    valid_noninteractive_options['description'] = "my description"
+    ## Specifically give a string value for interactive prompts
+    if not non_interactive:
+        valid_noninteractive_options['enabled'] = "True"
+    valid_noninteractive_options['transport'] = "local"
+    valid_noninteractive_options['scheduler'] = "direct"
+    valid_noninteractive_options['shebang'] = "#!/bin/bash"
+    valid_noninteractive_options['work-dir'] = "/scratch/{username}/aiida_run"
+    valid_noninteractive_options['mpirun-command'] = "mpirun -np {tot_num_mpiprocs}"
+    valid_noninteractive_options['mpiprocs-per-machine'] = "2"
+    # Make them multiline to test also multiline options
+    valid_noninteractive_options['prepend-text'] = "date\necho 'second line'"
+    valid_noninteractive_options['append-text'] =  "env\necho '444'\necho 'third line'"
+
+    # I replace kwargs here, so that if they are known, they go at the right order
+    for k in replace_args:
+        valid_noninteractive_options[k] = replace_args[k]
+
+    return valid_noninteractive_options
+
+def generate_setup_options(ordereddict):
+    """
+    Given an (ordered) dict, returns a list of options
+
+    Note that at this moment the implementation only supports long options
+    (i.e. --option=value) and not short ones (-o value).
+    Set a value to None to avoid the '=value' part.
+
+    :param ordereddict: as generated by ``generate_setup_options_dict()``
+    :return: a list to be passed as command-line arguments.
+    """
+    options = []
+    for k, v in ordereddict.iteritems():
+        if v is None:
+            options.append('--{}'.format(k))
+        else:
+            options.append('--{}={}'.format(k, v))
+    return options
+
+def generate_setup_options_interactive(ordereddict):
+    """
+    Given an (ordered) dict, returns a list of options
+
+    Note that at this moment the implementation only supports long options
+    (i.e. --option=value) and not short ones (-o value).
+    Set a value to None to avoid the '=value' part.
+
+    :param ordereddict: as generated by ``generate_setup_options_dict()``
+    :return: a list to be passed as command-line arguments.
+    """
+    options = []
+    for k, v in ordereddict.iteritems():
+        if v is None:
+            options.append(True)
+        else:
+            options.append(v)
+    return options
+
+class TestVerdiComputerSetup(AiidaTestCase):
+    """Tests for the 'verdi computer setup' command."""
+
+    def setUp(self):
+        self.runner = CliRunner()
+
+    def test_help(self):
+        self.runner.invoke(setup_computer, ['--help'])
+
+    def test_reachable(self):
+        import subprocess as sp
+        output = sp.check_output(['verdi', 'computer', 'setup', '--help'])
```

Comment:
Switch off exception catching here too, because 'Usage' will also be in the output if `computer` is considered an invalid command.

Reply:
This is check_output, that raises an exception in case of non-zero return code: https://docs.python.org/2/library/subprocess.html#subprocess.check_output
So I don't think I need to change here

Reply:
My bad...